### PR TITLE
Add user signup

### LINKED ISSUE #276: Stop Viewer not working in Middleware Deployments
When opening the stop viewer in middleware-enabled deployments,
the blank-screen-of-death is shown
and the below error message is shown in the browser console:
```log
react_devtools_backend.js:2273
TypeError: (0 , r.default)(...).tz is not a function
    at t.value (bundle.js:6)
    at ...
```

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR enables the user sign-in for deployments that feature OTP Middleware.

This PR also fixes #276.

Caveats right now:
- The nav bar has a thin black border (instead of light color), this is fixable in custom CSS (should there be no border at all instead?).
  * The quickest/easiest change for that is to set a transparent border color in the existing block in `style.scss`:
```css
.otp .navbar {
  background-color: $base-color;
  border-color: transparent;
  border-radius: 0;
}
```
- Appearance of the menu button and sign-in button with light-colored backgrounds may need to be addressed. EDIT: one deployment we considered involved a nav bar with a bright yellow background, and the white hamburger symbol and sign-in text will be difficult to read on such background. 
  * The colors for the sign-in button are [currently defined in OTP-RR](https://github.com/opentripplanner/otp-react-redux/blob/dev/lib/components/user/nav-login-button.css#L21). To override the colors, the two CSS blocks involved will need to be copied to `style.scss` and modified accordingly.
  * The color for the hamburger menu symbol can be overwritten in `style.scss` by adding this entry:
```css
.app-menu-button {
  color: <new color>;
}
```

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -1,17 +1,23 @@
 // import React/Redux libraries
 import React, { Component } from 'react'
 // import Bootstrap Grid components for layout
-import { Navbar, Grid, Row, Col } from 'react-bootstrap'
+import { Nav, Navbar, Grid, Row, Col } from 'react-bootstrap'
```
Comment (evansiroky) [inline lib/app.js:4]: `Nav`, `Navbar` no longer used in this file.

File: lib/app.js [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
   MobileMain,
+  NavLoginButtonAuth0,
   ResponsiveWebapp,
-  AppMenu
+  AppMenu,
+  otpUtils
 } from 'otp-react-redux'
 
+const { accountLinks } = otpUtils.auth
+
```
Comment (evansiroky) [inline lib/app.js:20]: `NavLoginButtonAuth0`, `AppMenu`, `accountLinks` (and therefore `otpUtils`) not used in this file.

**Review by evansiroky — APPROVED** [review]
Just some stuff that's no longer needed. If this project had a linter, then this would've been caught, but that's something for another PR.

**Review by landonreed — APPROVED** [review]
Looks good to me. What CSS is needed to change the nav border (please document in PR description)? Also, I don't really think the light-colored background for the nav menus is a problem... isn't that how they have always appeared?

**Comment by binh-dam-ibigroup** [comment]
> Looks good to me. What CSS is needed to change the nav border (please document in PR description)? Also, I don't really think the light-colored background for the nav menus is a problem... isn't that how they have always appeared?

@landonreed I updated the PR description for handling nav border and color changes. Let me know of other needed changes.

**Comment by binh-dam-ibigroup** [comment]
Proposing to block this PR until https://github.com/opentripplanner/otp-react-redux/pull/257 is released.

**Comment by landonreed** [comment]
@binh-dam-ibigroup, I just closed https://github.com/opentripplanner/otp-react-redux/pull/257. I think we should just modify the style in styles.scss instead of tweaking that specific bootstrap style for navbar-inverse. Please move forward with that approach.

**Comment by binh-dam-ibigroup** [comment]
If merging this PR, then styling from https://github.com/ibi-group/configurations/pull/96 should be used too.

**Pull request merged**